### PR TITLE
vm_xml.py: add seclable element to os nvram source part

### PR DESCRIPTION
**Pull request by meinaLi**
Add selable element to os nvram source part

### CODE REVIEW

**Comment by cliping** [comment]
In 'add selable element to os nvram source part', is 'selable' a misspelling? 'seclabel'?

**Review by cliping — APPROVED** [review]
Others LGTM

**Comment by meinaLi** [comment]
> In 'add selable element to os nvram source part', is 'selable' a misspelling? 'seclabel'?

Yes, just a typo. I've updated it. Thanks.

**Review by chloerh — APPROVED** [review]

File: virttest/libvirt_xml/vm_xml.py [hunk 1]
```diff
@@ -3227,6 +3232,32 @@ def __init__(self, virsh_instance=base.virsh):
                 virsh_instance=virsh_instance)
             self.xml = '<source/>'
 
+        @staticmethod
+        def marshal_from_seclabels(item, index, libvirtxml):
+            """
+            Convert an xml object to seclabel tag and xml element.
+            """
+            if isinstance(item, Seclabel):
```
Comment (chloerh) [inline virttest/libvirt_xml/vm_xml.py:3240]:
```suggestion
            if isinstance(item, librarian.get('seclabel')):
```

File: virttest/libvirt_xml/vm_xml.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from ..libvirt_xml import base, accessors, xcepts
 from ..libvirt_xml.base import LibvirtXMLBase
 from ..libvirt_xml.devices import librarian
+from ..libvirt_xml.devices.seclabel import Seclabel
```
Comment (chloerh) [inline virttest/libvirt_xml/vm_xml.py:15]:
```suggestion

```

File: virttest/libvirt_xml/vm_xml.py [hunk 3]
```diff
@@ -3227,6 +3232,32 @@ def __init__(self, virsh_instance=base.virsh):
...
 
+        @staticmethod
+        def marshal_from_seclabels(item, index, libvirtxml):
+            """
+            Convert an xml object to seclabel tag and xml element.
+            """
+            if isinstance(item, Seclabel):
+                return 'seclabel', item
+            elif isinstance(item, dict):
+                seclabel = Seclabel()
```
Comment (chloerh) [inline virttest/libvirt_xml/vm_xml.py:3243]:
```suggestion
                seclabel = librarian.get('seclabel')()
```

File: virttest/libvirt_xml/vm_xml.py [hunk 4]
```diff
@@ -3227,6 +3232,32 @@ def __init__(self, virsh_instance=base.virsh):
...
+                                             "instances, not a %s" % str(item))
+
+        @staticmethod
+        def marshal_to_seclabels(tag, new_treefile, index, libvirtxml):
+            """
+            Convert a seclabel tag xml element to an object of Seclabel.
+            """
+            if tag != 'seclabel':
+                return None
+            newone = Seclabel(virsh_instance=libvirtxml.virsh)
```
Comment (chloerh) [inline virttest/libvirt_xml/vm_xml.py:3257]:
```suggestion
            newone = librarian.get('seclabel')(virsh_instance=libvirtxml.virsh)
```

**Review by chloerh — COMMENTED** [review]
I'd like to suggest we remove the import of Seclabel and use librarian.get() instead like other device classes in vm_xml

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.ovmf_backed_nvram
JOB ID     : fd09e60783a92b0313a51807a4b505bbe3e28370
JOB LOG    : /var/lib/avocado/job-results/job-2023-10-09T22.03-fd09e60/job.log
 (1/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.default: PASS (37.88 s)
 (2/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.with_selinux_seclabel: PASS (35.24 s)
 (3/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.with_dac_seclabel: PASS (35.80 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-10-09T22.03-fd09e60/results.html
JOB TIME   : 109.60 s
```

File: virttest/libvirt_xml/vm_xml.py (hunk 1 above)
Comment (meinaLi) [reply virttest/libvirt_xml/vm_xml.py]: Changed to use librarian.get().

File: virttest/libvirt_xml/vm_xml.py (hunk 2 above)
Comment (meinaLi) [reply virttest/libvirt_xml/vm_xml.py]: Updated.

File: virttest/libvirt_xml/vm_xml.py (hunk 3 above)
Comment (meinaLi) [reply virttest/libvirt_xml/vm_xml.py]: Updated.

File: virttest/libvirt_xml/vm_xml.py (hunk 4 above)
Comment (meinaLi) [reply virttest/libvirt_xml/vm_xml.py]: Updated.

**Review by chloerh — APPROVED** [review]
Thanks for taking my suggestion.

**Pull request merged**